src/App.tsx

```diff
@@ -14,11 +15,97 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGoods: string[];
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    selectedGoods: [],
+  };
+
+  addProduct = (product: string) => {
+    this.setState((prevState: State): State => {
+      const { selectedGoods } = prevState;
+
+      return { selectedGoods: [...selectedGoods, product] };
+    });
+  };
+
+  deleteProduct = (product: string) => {
+    this.setState((state: State) : State => {
+      return { selectedGoods: state.selectedGoods.filter(item => item !== product) };
+    });
+  };
+
+  clearSelection = () => {
+    this.setState({ selectedGoods: [] });
+  };
+
+  render() {
+    const { selectedGoods } = this.state;
+    const { length } = selectedGoods;
+
+    const toBeForm = length >= 2 ? 'are' : 'is';
+    const whatIsSelected = selectedGoods.reduce((prev, item, i) => {
+      if (i === 0) {
+        return item;
+      }
+
+      if (i !== length - 1) {
+        return `${prev}, ${item}`;
+      }
+
+      return `${prev} and ${item}`;
+    }, '');
+
+    return (
+      <div className="app">
+        <h1 className="app__heading">
+          {length > 0 ? `${whatIsSelected} ${toBeForm} selected` : 'Nothing is selected'}
+        </h1>
+
+        {selectedGoods.length > 0 && (
```

Comment:
you can turn any value to it's boolean equivalent by using `!!` operator. not that it will reduce the amount of code or anything, but i'll just leave it here

Reply:
the idea to use `!! ` just didn't cross my mind

Reply:
так якщо буде !!0 то це буде false і тобі не потрібно буде робити перевірки на 0
Якщо довжина буде більше чим 1 то !!1 поверне true